File: 2. Regression/1. Simple Linear Regression/simple_linear_regression.R
dataset = read.csv('Salary_Data.csv')

library(caTools)

split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

regressor = lm(formula = Salary ~ YearsExperience, data = training_set)
#summary(regressor)

y_pred = predict(regressor, newdata = test_set)

# install.packages('ggplot2')
library(ggplot2)

visualize_result <- function(X, y, title, xlabel, ylabel){
  ggplot() +
    geom_point(aes(x = X, y = y),
              colour = 'red') +
    geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
              colour = 'blue') +
    ggtitle(title) +
    xlab(xlabel) +
    ylab(ylabel)
}

simple_linear_regression_visualization <- function(X, y){
  visualize_result(X, y, 'Salary vs Experience', 'Years of experience', 'Salary')
}

visualize_result(training_set$YearsExperience, training_set$Salary)
visualize_result(test_set$YearsExperience, test_set$Salary)

File: 2. Regression/1. Simple Linear Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, -1].values

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3)


from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

y_pred = regressor.predict(X_test)

def visulize_result(X, y, title, xlabel, ylabel):
    plt.scatter(X_train, y_train, color='red', marker='x')
    plt.plot(X_train, regressor.predict(X_train), color='blue')
    plt.title()
    plt.xlabel("Years of Experience")
    plt.ylabel("Salary")
    plt.show()

def simple_linear_regression_visulization(X, y):
    return visulize_result(X, y, "Salary vs Experience", "Years of Experience", "Salary")

simple_linear_regression_visulization(X_train, y_train)
simple_linear_regression_visulization(X_test, y_test)